setup.py

```diff
@@ -39,10 +42,32 @@
     install_and_import("pybind11[global]")
     from pybind11.setup_helpers import build_ext as BuildExtension
 
+# Start timing
+start_time = time.perf_counter()
+
+
+def print_total_time():
+    end_time = time.perf_counter()
+    total_time = end_time - start_time
+    print(f"Total build time: {total_time:.2f} seconds")
+
+
+# Register the print_total_time function to be called at exit
+atexit.register(print_total_time)
```

Comment:
Could this give bogus results if the user builds a downstream dependency?
```
$ python -m pip install other-package
Collecting flash_attn
Collecting transformer_engine
Collecting other_package
Total build time: <transformer_engine + other_package build time>
```
If we keep this, we should make sure Pip builds each package within a separate Python process.

Reply:
It does not print times without verbose mode so I don't think users will have such trouble.

Reply:
I replace the print `atexit` with a normal Python print, to avoid unexpected behaviours.